### PR TITLE
cmake option: `CUDF_KVIKIO_REMOTE_IO`

### PR DESCRIPTION
**Pull request by madsbk**
Compile flag to enable/disable remote IO through KvikIO: `CUDF_KVIKIO_REMOTE_IO`

### CODE REVIEW

File: cpp/CMakeLists.txt
Comment (ttnghia) [inline cpp/CMakeLists.txt:95]: If this is not enable, the C++ code will not support it thus I don't think Python will be able to do it.

File: cpp/src/io/utilities/datasource.cpp
Comment (ttnghia) [inline cpp/src/io/utilities/datasource.cpp:394]: Please also wrap the corresponding header by this macro too.

File: cpp/src/io/utilities/datasource.cpp
Comment (ttnghia) [inline cpp/src/io/utilities/datasource.cpp:481]: Remove parameter name to avoid unused warning.
```suggestion
  static constexpr bool is_supported_remote_url(std::string const&) { return false; }
```

File: cpp/CMakeLists.txt
Comment (madsbk) [reply cpp/CMakeLists.txt]: Updated: 
```cmake

option(
  CUDF_KVIKIO_REMOTE_IO
  "Enable remote IO (e.g. AWS S3) support through KvikIO. If disabled, cudf-python will still be able to do remote IO through fsspec."
  ON
)

```

File: cpp/src/io/utilities/datasource.cpp
Comment (madsbk) [reply cpp/src/io/utilities/datasource.cpp:397]: done

**Comment by madsbk** [comment]
@ttnghia, can you check if setting `-DCUDF_KVIKIO_REMOTE_IO=OFF` fixes fixes https://github.com/NVIDIA/spark-rapids-jni/issues/2580 ?

**Comment by pxLi** [comment]
it would be nice to also include an update to cudf JNI build script, thanks!
https://github.com/rapidsai/cudf/blob/branch-24.12/java/ci/build-in-docker.sh#L68

**Comment by ttnghia** [comment]
> @ttnghia, can you check if setting `-DCUDF_KVIKIO_REMOTE_IO=OFF` fixes fixes [NVIDIA/spark-rapids-jni#2580](https://github.com/NVIDIA/spark-rapids-jni/issues/2580) ?

Indeed https://github.com/NVIDIA/spark-rapids-jni/issues/2580 should be fixed by https://github.com/rapidsai/kvikio/pull/542.
For this PR, I confirm that it fixes the linking issue https://github.com/NVIDIA/spark-rapids-jni/issues/2581. Thanks for working on this.

**Review by kingcrimsontianyu — APPROVED** [review]
Lgtm. Thanks for the fix!

**Comment by pxLi** [comment]
Still requires approval from `cudf-cmake-codeowners`, thanks

**Review by bdice — APPROVED** [review]

**Pull request merged**
